src/SearchParameters/index.js

```diff
@@ -1327,17 +1330,32 @@ SearchParameters.prototype = {
       throw error;
     }
 
-    var parsedParams = SearchParameters._parseNumbers(params);
-
-    return this.mutateMe(function mergeWith(newInstance) {
-      var ks = keys(params);
-
-      ks.forEach(function(k) {
-        newInstance[k] = parsedParams[k];
-      });
+    var self = this;
+    var nextWithNumbers = SearchParameters._parseNumbers(params);
+    var previousPlainObject = Object.keys(this).reduce(function(acc, key) {
+      acc[key] = self[key];
+      return acc;
+    }, {});
+
+    var nextPlainObject = Object.keys(nextWithNumbers).reduce(
+      function(previous, key) {
+        var isPreviousValueDefined = previous[key] !== undefined;
+        var isNextValueDefined = nextWithNumbers[key] !== undefined;
+
+        if (isPreviousValueDefined && !isNextValueDefined) {
+          return omit(previous, [key]);
+        }
+
+        if (isNextValueDefined) {
+          previous[key] = nextWithNumbers[key];
+        }
+
+        return previous;
+      },
+      previousPlainObject
```

Comment:
I wonder if we could have `this` here, then I don't think we need to call new this.constructor anymore, but rather return `this` & ignore the value returned from the reduce (ie use forEach instead)

Reply:
We have to call the constructor, we don't want to mutate the `SearchParameters`. We have to return a new instance. The `mutateMe` function despite the name was returning [a new instance](https://github.com/algolia/algoliasearch-helper-js/pull/703/files/4942fbd413afb02cb3b3a17a185c68eb2129422b#diff-ecd72976141723d9448df45e667cb4beL1376) of the `SearchParameters`. With `this` the only solution to "omit" attributes is the usage of the `delete` operator (which might be a bit slower). Without talking about performance I'm more comfortable with this implementation.